ql/src/java/org/apache/hadoop/hive/ql/session/SessionState.java

```diff
@@ -765,6 +768,32 @@ private static void start(SessionState startSs, boolean isAsync, LogHelper conso
     }
   }
 
+  /**
+   * Set hive metastore delegation token in current UGI
+   */
+  private void setHiveMetastoreDelegationTokenToUGI() {
+    if (this.isHiveServerQuery) {
+      return;
+    }
+    if (!getSessionConf().getBoolVar(ConfVars.METASTORE_USE_THRIFT_SASL)) {
+      return;
+    }
+    try {
+      UserGroupInformation UGI = Utils.getUGI();
+      String hmsDelegationTokenStr = Hive.get().getDelegationToken(UGI.getShortUserName(), UGI.getShortUserName());
+      if (hmsDelegationTokenStr != null) {
+        getSessionConf().setVar(ConfVars.METASTORE_TOKEN_SIGNATURE, CLITOKEN);
+        try {
+          SecurityUtils.setTokenStr(UGI, hmsDelegationTokenStr, CLITOKEN);
```

Comment:
In HiveSessionImplwithUGI, we set the HiveServer2ImpersonationToken token before the SessionState starts, will the HiveClientImpersonationToken override the HiveServer2ImpersonationToken?

Reply:
> In HiveSessionImplwithUGI, we set the HiveServer2ImpersonationToken token before the SessionState starts, will the HiveClientImpersonationToken override the HiveServer2ImpersonationToken?

If hiveserverquery,will return directly

```
private void setHiveMetastoreDelegationTokenToUGI() {
    if (this.isHiveServerQuery) {
      return;
    }
...
}
```

Reply:
Got it, thank you!